lib_xua/src/core/mixer/mixer.xc

```diff
@@ -737,6 +810,40 @@ static void mixer2(chanend c_mixer1)
 #endif
 #endif
         }
+
+    // mixer2 is responsible for output volume control, unless the sample is not available with mixer2,
+    // in which case it leaves it to mixer1 to do.
+    // mixer2 has ptr_samples[0..(NUM_USB_CHAN_OUT+NUM_USB_CHAN_IN)] as well as index 1,3,5 and 7 (so the odd indexes) of the mixed outputs
+
+#if (OUT_VOLUME_IN_MIXER && OUT_VOLUME_AFTER_MIX)
+        #warning OUT Vols in mixer, AFTER mix & map
+        #pragma loop unroll
+        for (int i=0; i<NUM_USB_CHAN_OUT; i++)
+        {
+            int ch_index;
+            unsafe {
+            ch_index = samples_to_device_map[i];
+            }
+            int do_vol_control = 0;
```

Comment:
int do_vol_control = (ch_index < NUM_USB_CHAN_OUT + NUM_USB_CHAN_IN);

simpler?

Reply:
I think the longer form explains better when exactly volume control is done? I add some comments also
```
            int do_vol_control = 0;
            if(ch_index < (NUM_USB_CHAN_OUT + NUM_USB_CHAN_IN)) /* if ch_index is one of the usb out or in channels*/
            {
                do_vol_control = 1;
            }
            else
            {
                if((ch_index - (NUM_USB_CHAN_OUT + NUM_USB_CHAN_IN)) & 0x1) /* if ch_index is an odd numbered mixer output channel*/
                {
                    do_vol_control = 1;
                }
            }
```